File: clients/node/examples/workflow-connect.js
import { Screen, Pattern } from "../src";

async function main() {
  // Auto-launch workflow.
  const screen = await Screen.auto();
  try {
    const match = await screen.click(new Pattern("assets/pattern.png").exact());
    console.log(`clicked match target at (${match.targetX}, ${match.targetY})`);
  } finally {
    await screen.close();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: clients/python/examples/workflow_connect.py
from __future__ import annotations

import os

from sikuligo import Pattern, Screen

screen = Screen.connect(address=os.getenv("SIKULI_GRPC_ADDR", "127.0.0.1:50051"))
try:
    match = screen.click(Pattern("assets/pattern.png").exact())
    print(f"clicked match target at ({match.target_x}, {match.target_y})")
finally:
    screen.close()
